zeppelin-interpreter/src/test/java/org/apache/zeppelin/interpreter/RemoteZeppelinServerResourceTest.java

```diff
@@ -0,0 +1,150 @@
+                "}"), serializedJSON);
+
+        // Now, let's check the serializedJson deserialization
+        RemoteZeppelinServerResource deserialized = RemoteZeppelinServerResource.fromJson(serializedJSON);
+        assertEquals(original.getOwnerKey(), deserialized.getOwnerKey());
+        assertEquals(original.getResourceType(), deserialized.getResourceType());
+        assertEquals(original.getData(), deserialized.getData());
+    }
+
+    @Test
+    void fromJson_withInvalidResourceType() {
+        String invalidJson = a2q("{" +
+                "'ownerKey':'1234'," +
+                "'resourceType':'INVALID_TYPE'," +
+                "'data':{'key1':'value','key2':'value2'}" +
+                "}");
+
+        RemoteZeppelinServerResource deserialized = RemoteZeppelinServerResource.fromJson(invalidJson);
+        // Deserialization is successful and type is null
+        assertNull(deserialized.getResourceType());
+        // Owner key and data are deserialized correctly
+        assertEquals("1234", deserialized.getOwnerKey());
+        assertEquals(mapOf("key1", "value", "key2", "value2"), deserialized.getData());
+    }
+
+    @Test
+    void fromJson_withNothing() {
+        // When only the empty JSON object is provided,
+        RemoteZeppelinServerResource deserialized = RemoteZeppelinServerResource.fromJson("{}");
+
+        // Then, all fields should be null
+        assertNull(deserialized.getOwnerKey());
+        assertNull(deserialized.getResourceType());
+        assertNull(deserialized.getData());
+    }
+
+    @Test
+    void fromJson_withNullFields() {
+        String jsonWithNullFields = a2q("{" +
+                "'ownerKey':null," +
+                "'resourceType':null," +
+                "'data':null" +
+                "}");
+
+        RemoteZeppelinServerResource deserialized = RemoteZeppelinServerResource.fromJson(jsonWithNullFields);
+        assertNull(deserialized.getOwnerKey());
+        assertNull(deserialized.getResourceType());
+        assertNull(deserialized.getData());
+    }
+
+    @Test
+    void toJson_withEmptyData() {
+        RemoteZeppelinServerResource original = new RemoteZeppelinServerResource();
+        original.setOwnerKey("");
+        original.setData(new TreeMap<>());
+        original.setResourceType(PARAGRAPH_RUNNERS);
+
+        String serializedJSON = original.toJson();
+        assertEquals(a2q("{" +
+                "'ownerKey':''," +
+                "'resourceType':'PARAGRAPH_RUNNERS'," +
+                "'data':{}" +
+                "}"), serializedJSON);
+
+        RemoteZeppelinServerResource deserialized = RemoteZeppelinServerResource.fromJson(serializedJSON);
+        assertEquals(original.getOwnerKey(), deserialized.getOwnerKey());
+        assertEquals(original.getResourceType(), deserialized.getResourceType());
+        assertEquals(original.getData(), deserialized.getData());
+    }
+
+    @Test
+    void fromJson_withLargeDataSet() {
```

Comment:
It seems like a performance test. In my opinion, it's better to avoid performance testing in unit tests. I think unit tests and performance tests have different purposes.

Reply:
Oh my bad. I think the test name made you get the wrong idea 😅.
The test is intended to test (sort of) long data as input.
There is a whole lot of Zeppelin integration tests that run wayyyy over a minute 😆
This test won't even leave a dent on the performance of Zeppelin test suite.

I changed the name from `LargeDataSet` to `LongDataset`.

Reply:
@JooHyukKim 
Does using long data actually change the functionality here? If it's just a unit test, the `toJson` test in this class already seems to cover the same function.